src/hotspot/share/jfr/recorder/service/jfrEventThrottler.cpp

```diff
@@ -69,15 +75,19 @@ void JfrEventThrottler::destroy() {
 // and another for the SamplingLatency event.
 // When introducing many more throttlers, consider adding a lookup map keyed by event id.
 JfrEventThrottler* JfrEventThrottler::for_event(JfrEventId event_id) {
+  assert(_disabled_cpu_time_sample_throttler != nullptr, "Disabled CPU time throttler has not been properly initialized");
   assert(_object_allocation_throttler != nullptr, "ObjectAllocation throttler has not been properly initialized");
   assert(_safepoint_latency_throttler != nullptr, "SafepointLatency throttler has not been properly initialized");
-  assert(event_id == JfrObjectAllocationSampleEvent || event_id == JfrSafepointLatencyEvent, "Event type has an unconfigured throttler");
+  assert(event_id == JfrObjectAllocationSampleEvent || event_id == JfrSafepointLatencyEvent || event_id == JfrCPUTimeSampleEvent, "Event type has an unconfigured throttler");
   if (event_id == JfrObjectAllocationSampleEvent) {
     return _object_allocation_throttler;
   }
   if (event_id == JfrSafepointLatencyEvent) {
     return _safepoint_latency_throttler;
   }
+  if (event_id == JfrCPUTimeSampleEvent) {
+    return _disabled_cpu_time_sample_throttler;
```

Comment:
Why do you need to make a "disabled" throttler? Shouldn't the rate limiting always be done by the arguments given to `timer_create`? The actual JFR throttling mechanism used by `ObjectAllocationSample` doesn't really fit this use-case anyway since you can already pre-determine the rate at which events are emitted (while with `ObjectAllocationSample` you can't).

Reply:
This just makes the code simpler and avoids two additional checks on the throttler hot-path. The _disabled_cpu_time_sample_throttler is like `nullptr` but without the `nullptr` check.